src/components/events/eventFeed/eventFeed.tsx

```diff
@@ -18,33 +18,29 @@ import {
 import { userRole } from '../../../types/string-types';
 
 interface eventProps {
-    eventId: string | undefined;
     userRole: userRole;
+    handleAlert: (isSuccess: boolean, content: string) => void;
 }
 
-const Event = ({ eventId, userRole }: eventProps) => {
+const Event = ({ userRole, handleAlert }: eventProps) => {
   const [event, setEvent] = useState<IEvent>();
   const [loading, setLoading] = useState<boolean>(false);
   const [isEdit, setIsEdit] = useState<boolean>(false);
-
-  if (!eventId) {
-    const navigate = useNavigate();
-    navigate('/events');
-    return null;
-  }
+  const { id } = useParams();
 
   const fetchEvent = useCallback(async (): Promise<void> => {
-    const { status, data } = await getEventById(eventId);
-    if (status !== StatusCodes.OK) {
-      throw new Error('Error! Event not found');
+    if (id) {
+      const { status, data } = await getEventById(id);
+      if (status !== StatusCodes.OK) {
+        throw new Error('Error! Event not found');
+      }
+      setEvent(data.event);
+      setLoading(false);
     }
-    setEvent(data.event);
-  }, [eventId]);
+  }, []);
```

Comment:
are you aware that any change in the data will not fire another req?

Reply:
Yes.
I expect no changes in "id" in the life cycle of this component because it is per url.
Once I entered this url the only event Ill display is the one in the url, and it cannot change while the url does not change.